regex/src/main/scala/regex/Regex.scala

```diff
@@ -15,13 +22,41 @@ package regex
   */
 
 /** Simplifies a regular language */
-def simplify(lang: RegularLanguage): RegularLanguage = ???
+def simplify(lang: RegularLanguage): RegularLanguage = lang match
+  case Concat(re1, re2) if re1 == Epsilon => re2
```

Comment:
Instead of using if statements, I did pattern matching which might be easier to read. For example, I would have
`case Concat(Epsilon, r: RegularLanguage) => simplify(r)
  case Concat(r: RegularLanguage, Epsilon) => simplify(r)
  case Concat(Empty, r: RegularLanguage) => simplify(Empty)
  case Concat(r: RegularLanguage, Empty) => simplify(Empty)`